src/server/engine_factory.rs

```diff
@@ -167,6 +168,9 @@ impl KvEngineFactory {
         };
         let shared_block_cache = self.inner.block_cache.is_some();
         kv_engine.set_shared_block_cache(shared_block_cache);
+        // RocksEngine is there now, we can set its write compaction filter factory
+        DbConfig::set_write_compaction_filter_factory(kv_engine.clone(), kv_cfs_opts);
```

Comment:
I'm not sure it will work by this way. The engine is created already, and maybe the set_write_compaction_filter_factory cannot not take effect.